multimodal/src/autogluon/multimodal/models/sam.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import os
+import time
+import warnings
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+from omegaconf import DictConfig
+from torch import nn
+from transformers import SamConfig, SamModel
+
+from ..constants import COLUMN, IMAGE, IMAGE_VALID_NUM, LABEL, LOGITS, SEMANTIC_SEGMENTATION
+from .utils import assign_layer_ids, freeze_model_layers
+
+logger = logging.getLogger(__name__)
+
+
+class SAMForSemanticSegmentation(nn.Module):
+    """
+    Support SAM for binary real-world semantic segmentation.
+    Refer to https://huggingface.co/docs/transformers/main/model_doc/sam
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str,
+        config: DictConfig,
+        pretrained: Optional[bool] = True,
+        frozen_layers: Optional[list] = None,
+    ):
+        """
+        Load a pretrained Segment Anything Model (SAM).
+
+        Parameters
+        ----------
+        prefix
+            The prefix of the SAMForSemanticSegmentation model.
+        checkpoint_name
+            Name of the SAM checkpoint.
+        pretrained
+            Whether using the pretrained SAM models. If pretrained=True, download the pretrained model.
+        frozen_layers
+            A list of substrings of frozen layers' names.
+        """
+
+        super().__init__()
+        self.prefix = prefix
+        self.pretrained = pretrained
+        self.checkpoint_name = checkpoint_name
+        self.frozen_layers = frozen_layers
+
+        self.device = None
+        self.name_to_id = {}
+
+        self._load_checkpoint(checkpoint_name)
+
+        freeze_model_layers(self.model, self.frozen_layers)
```

Comment:
What's the relation of frozen layers and peft? Will we still insert lora layers in the frozen layers?